### PR TITLE
Homework3: Shevchuk Aleksei

### PR DESCRIPTION
**Pull request by alex-shef**

### CODE REVIEW

**Comment by alex-shef** [comment]
1st PR

File: 2nd lecture/Homework.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        if n % i == 0:
+            s += 1
+    if s == 2:
+        print('простое')
+    else:
+        print('составное')
+
+# ***Проверьте, является ли число палиндромом
+
+s = 0
```
Comment (belovmd) [inline 2nd lecture/Homework.py]: it's not completely clear what these variables keep: s m n

File: 2nd lecture/Homework.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+n = int(input('Введите целое число: ', ))
+for i in range(1, n + 1):
+    print(i ** 3, '')
+
+# Проверьте, является ли введенное число простым.
+
+if n > 1:
+    s = 0
+    for i in range(1, n + 1):
```
Comment (belovmd) [inline 2nd lecture/Homework.py]: it would be enough int(n**0,5)+1

File: 2nd lecture/Homework.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    print('палиндром')
+
+n = str(n)
+if n == n[::-1]:
+    print('палиндром')
+
+# ***Выведите первые n чисел Фибоначчи,
+# используйте временные переменные для хранения прошлых значений и цикл for
+
+n = int(n)
```
Comment (belovmd) [inline 2nd lecture/Homework.py]: please separate inputs for different tasks and add invitation strings

File: 2nd lecture/Homework.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+f = 0
```
Comment (belovmd) [inline 2nd lecture/Homework.py]: unclear variables namings

File: 2nd lecture/Homework.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+if n == 1:
+    print(0)
+elif n == 2:
+    print(0)
+    print(1)
+else:
+    print(0)
+    print(1)
+    for i in range(1, n - 1):
+        f = f1 + f0
```
Comment (belovmd) [inline 2nd lecture/Homework.py]: num_prev, num_before_prev = num1+num2, num1

File: 3rd lecture/Homework2.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+# https://acmp.ru/index.asp?main=task&id_task=384
+# Числа Фибоначчи - 3
+# Требуется найти наибольший общий делитель двух чисел Фибоначчи.
+
+i = int(input("2. Введите индексы чисел Фибоначчи: ", ))
+j = int(input("Введите индексы чисел Фибоначчи: ", ))
+
+
+def fibo(x):
```
Comment (belovmd) [inline 3rd lecture/Homework2.py]: can it count 50th fibonacci number?

File: 3rd lecture/Homework2.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        a = a - b
+    else:
+        b = b - a
+
+print("НОД=", a)
+
+# 3. Найти самое длинное слово в введенном предложении.
+# **: учтите что в предложении есть знаки препинания.
+
+a = input('3. Введите предложение: ', )
```
Comment (belovmd) [inline 3rd lecture/Homework2.py]: what about `;()[]{}' etc? it seems split would be better

File: 3rd lecture/Homework2.py [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    if len(elem) > len(maxlen):
+        maxlen = elem
+print("Самое длинное слово:", maxlen)
+
+# 4. Вводится строка.
+# Требуется удалить из нее повторяющиеся символы и все пробелы.
+# Например, если было введено "abc cde def", то должно быть выведено "abcdef".
+
+a = input('4. Введите строку: ', )
+a = a.replace(' ', '')
```
Comment (belovmd) [inline 3rd lecture/Homework2.py]: a not best naming

**Review by belovmd — COMMENTED** [review]
1. please remove tasks from our lessons
2. please remove tasks from 1st homework
3. Some tasks are duplicated(palindrom, fibonacci)
4. Please don't use russian letters in text

**Pull request closed without merging**